server/bleep/src/semantic.rs

```diff
@@ -424,7 +540,24 @@ impl Semantic {
                     .map(Payload::from_qdrant)
                     .collect::<Vec<_>>()
             })?;
-        Ok(deduplicate_snippets(results, vector, limit))
+        let dedup_results = deduplicate_snippets(results, vector.clone(), limit);
+        let results_lexical = self
+            .search_lexical(parsed_query, vector.clone(), limit, offset, 0.0)
+            .await
+            .map(|raw| {
+                raw.into_iter()
+                    .map(Payload::from_qdrant)
+                    .collect::<Vec<_>>()
+            })?;
+        let results_lexical = Self::rank_lexical(results_lexical, &query);
+
+        let merged_results = Self::merge_rrf(results_lexical, dedup_results);
+
+        Ok(merged_results
+            .iter()
+            .take(limit.try_into().unwrap())
+            .cloned()
```

Comment:
can consume here instead of cloning?
```suggestion
            .into_iter()
            .take(limit.try_into().unwrap())
```